src/services/package-manager/batch-import-export-service.ts

```diff
@@ -0,0 +1,38 @@
+import {batchImportExportApi} from "../../api/batch-import-export-api";
+import {logger} from "../../util/logger";
+import {v4 as uuidv4} from "uuid";
+import {PackageExportTransport} from "../../interfaces/package-export-transport";
+import {FileService, fileService} from "../file-service";
+import {studioService} from "../studio/studio.service";
+
+class BatchImportExportService {
+
+    public async listActivePackages(flavors: string[]): Promise<void> {
+        const activePackages = await batchImportExportApi.findAllActivePackages(flavors);
+        activePackages.forEach(pkg => {
+            logger.info(`${pkg.name} - Key: "${pkg.key}"`)
+        });
+    }
+
+    public async findAndExportListOfActivePackages(flavors: string[], packageKeys: string[], withDependencies: boolean): Promise<void> {
+        let packagesToExport: PackageExportTransport[];
+
+        if (packageKeys.length) {
+            packagesToExport = await batchImportExportApi.findActivePackagesByKeys(packageKeys, withDependencies);
+        } else  {
+            packagesToExport = await batchImportExportApi.findAllActivePackages(flavors, withDependencies);
+        }
+
+        packagesToExport = await studioService.getExportPackagesWithStudioData(packagesToExport, withDependencies);
+
+        this.exportListOfPackages(packagesToExport);
+    }
+
+    private exportListOfPackages(packages: PackageExportTransport[]): void {
```

Comment:
Have you taken into consideration to actually export this as a ZIP file with each package having its own JSON file?
I'm not really clear on the feature yet, but having a long list of packages here can make it unreadable

Reply:
No, we did not consider that. We wanted to make this similar to listing other resources (e.g. packages, data-pools) which are also exported as json.